releng/com.b2international.snowowl.server.update/assembly/common/scripts/snowowl_delta_import.sh

```diff
@@ -0,0 +1,244 @@
+SNOW_OWL_USER=""
+
+# The password for the above user
+SNOW_OWL_USER_PASSWORD=""
+
+# The source folder with the exported RF2 content
+SOURCE_FOLDER=""
+
+#
+# Global variables / constants, advanced parameters
+#
+
+# Base URL of the Snow Owl server
+SNOW_OWL_BASE_URL="http://localhost:8080"
+
+# URL for Snow Owl's REST API
+SNOW_OWL_API_URL="/snowowl/snomed-ct/v3"
+
+# Media type to use for REST requests
+MEDIA_TYPE="application/vnd.com.b2international.snowowl+json"
+
+# The url of the bugzilla repository, e.g.: http://localhost:9090/bugzilla
+BUGZILLA_URL=""
+
+#Login credentials to bugzilla
+BUGZILLA_USERNAME=""
+BUGZILLA_PASSWORD=""
+
+usage() {
+
+	cat <<EOF
+
+NAME:
+
+    Delta import script for the Snow Owl terminology server
+
+    OPTIONS:
+
+    -h
+        Show this help
+    -u
+        Define a username with privileges to the Snow Owl REST API
+    -p
+        Define the password for the above user
+    -b
+        Snow Owl base URL, defaults to 'http://localhost:8080'
+    -f
+        Source folder where the exported content can be found
+    -s
+        Branch from Snow Owl to initiate the import on
+    -a
+        REST API URL of the Snow Owl server, defaults to '/snowowl/snomed-ct/v3'
+	-z
+		Bugzilla URL
+	-U
+		Defines the bugzilla user
+	-P
+		Defines the bugzilla pasword for the above user
+
+NOTES:
+
+    This script can be used to import content to a Snow Owl terminology server.
+
+    Mandatory variables:
+        - SNOW OWL user that is able to authenticate through the REST API
+        - Password for the above user
+        - Source folder with the delta export files
+		- Bugzilla URL
+		- Bugzilla user
+		- Bugzilla password for the above user
+
+EOF
+
+}
+
+echo_date() {
+	echo -e "[$(date +"%Y-%m-%d %H:%M:%S")] $@"
+}
+
+check_if_empty() {
+	if [[ -z "$1" ]]; then
+		echo_date "$2"
+		exit 1
+	fi
+}
+
+validate_variables() {
+	check_if_empty "${SNOW_OWL_USER}" "Snow Owl username must be specified"
+	check_if_empty "${SNOW_OWL_USER_PASSWORD}" "User password must be specified"
+	check_if_empty "${SOURCE_FOLDER}" "Source folder must be specified"
+	SOURCE_FILE=$(ls -t snow_owl_DELTA_export* | head -1)
+	check_if_empty "${SOURCE_FILE}" "Source folder must contain a valid delta export file"
+	check_if_empty "${BUGZILLA_URL}" "Bugzilla url must be specified"
+	check_if_empty "${BUGZILLA_USERNAME}" "Bugzilla username must be specified"
+	check_if_empty "${BUGZILLA_PASSWORD}" "Bugzilla pasword must be specified"
+}
+
+bugzilla_login() {
+	BUGZILLA_LOGIN_ENDPOINT="${BUGZILLA_URL}/rest/login?login=${BUGZILLA_USERNAME}&password=${BUGZILLA_PASSWORD}"
+	BUGZILLA_TOKEN=$(curl --request GET \
+		--include --silent --show-error \
+		"${BUGZILLA_LOGIN_ENDPOINT}" | awk -F'"' '{print $6}')
+}
+
+bugzilla_lougout() {
+	BUGZILLA_LOGOUT_ENDPOINT="${BUGZILLA_URL}/rest/logout?token=${BUGZILLA_TOKEN}"
+	BUGZILLA_LOGOUT_ENDPOINT="$(echo -e "${BUGZILLA_LOGOUT_ENDPOINT}" | tr -d '[:space:]')"
```

Comment:
Is this filtering step still needed? It is only present in `bugzilla_logout` and `create_bug` but not in `bugzilla_login`.